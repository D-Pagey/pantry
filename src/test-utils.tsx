import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
