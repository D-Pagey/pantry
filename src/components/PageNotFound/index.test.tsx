import React from 'react';
import { render } from '../../test-utils';
import { PageNotFound } from '.';

describe('PageNotFound component', () => {
    it('should render', () => {
        const { container } = render(<PageNotFound />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
