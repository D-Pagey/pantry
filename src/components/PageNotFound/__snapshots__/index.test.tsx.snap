// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.06);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 24px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 2;
}

.c3 {
  font-size: 25px;
  font-weight: 500;
  margin: 0;
  padding: 0 0 0 14px;
}

.c2 {
  cursor: pointer;
  height: 16px;
  width: 16px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: transparent;
  border: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.c4 {
  position: relative;
  right: 4rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="arrow"
      class="c2"
      data-testid="headerBackArrow"
      src="arrow.svg"
    />
    <h4
      class="c3"
    >
      Not found
    </h4>
  </div>
  <div
    class="c4"
  >
    <button
      class="c5"
      type="button"
    >
      <svg
        fill="none"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.2222 28.8C22.2222 29.3835 21.9881 29.9431 21.5713 30.3556C21.1546 30.7682 20.5894 31 20 31C19.4106 31 18.8454 30.7682 18.4287 30.3556C18.0119 29.9431 17.7778 29.3835 17.7778 28.8H22.2222ZM20 9C20.2947 9 20.5773 9.11589 20.7857 9.32218C20.994 9.52847 21.1111 9.80826 21.1111 10.1V11.288C24.2667 11.816 26.6667 14.533 26.6667 17.8V24.4L30 27.7H10L13.3333 24.4V17.8C13.3333 14.533 15.7333 11.816 18.8889 11.288V10.1C18.8889 9.80826 19.006 9.52847 19.2143 9.32218C19.4227 9.11589 19.7053 9 20 9Z"
          fill="#686B6F"
        />
      </svg>
    </button>
  </div>
</div>
`;
